selene/src/main.rs

```diff
@@ -517,17 +530,23 @@ fn start(mut options: opts::Options) {
                 }
             }
 
-            None => match fs::read_to_string("selene.toml") {
-                Ok(config_contents) => match toml::from_str(&config_contents) {
-                    Ok(config) => (config, None),
-                    Err(error) => {
-                        error!("Config file not in correct format: {}", error);
-                        std::process::exit(1);
-                    }
-                },
+            None => {
+                let read_config_file = |file: &str| -> Option<CheckerConfig<toml::value::Value>> {

```

Comment:
Since finding the config file isn't so trivial anymore we should move the one from validate config command to a separate function and use that instead.

Reply:
refactored in https://github.com/Zeioth/selene/commit/13e628625f41b99d38aa817fd9c94a33a2e8b1ed

## More info
The code to read and validate a config file is now a separated function.
```rust
fn read_config_file(file: &str) -> Option<CheckerConfig<toml::value::Value>> {
    let config_contents = fs::read_to_string(file).ok()?;
    toml::from_str(&config_contents)
        .map_err(|error| {
            error!("Error parsing config file {}: {}", file, error);
        })
        .ok()
}
```

Reply:
I meant more like a separate function to find the config file. It’s needed in the validate config command logic as well: https://github.com/Kampfkarren/selene/pull/591#discussion_r1519041670

Reply:
@chriscerie ok this should do it:

* All logic regarding reading and parsing config files has been moved to dedicated functions:

```rust
/// Reads a config file and return content and path. None if not found or empty.
fn read_config_file() -> Option<(String, Option<PathBuf>)> {
    let config_paths_to_check = [".selene.toml", "selene.toml"];
    let mut config_contents = String::new();
    let mut config_path = None;

    for path in &config_paths_to_check {
        if let Ok(contents) = fs::read_to_string(path) {
            config_contents = contents;
            config_path = Some(PathBuf::from(path));
            break;
        }
    }

    if config_contents.is_empty() {
        return None;
    }

    Some((config_contents, config_path))
}

/// Read the config file and return it as Option<(String, &'static Path)>.
fn parse_config_file_as_string() -> Option<(String, &'static Path)> {
    if let Some((config_contents, config_path)) = read_config_file() {
        if let Some(path) = config_path {
            return Some((config_contents, Box::leak(Box::new(path)).as_path()));
        }
    } else {
        error!("Error reading config file");
        std::process::exit(1);
    }
    None
}

/// Read the config file and return it as CheckerConfig<toml::Value>, Option<PathBuf>.
fn parse_config_file_as_checkerconfig() -> Option<(CheckerConfig<toml::Value>, Option<PathBuf>)> {
    if let Some((config_contents, config_path)) = read_config_file() {
        match toml::from_str(&config_contents) {
            Ok(config) => Some((config, config_path)),
            Err(error) => {
                error!("Error parsing config file: {}", error);
                std::process::exit(1);
            }
        }
    } else {
        None
    }
}
```

The calls now look like:
```
let (config_content, config_path) = parse_config_file_as_string().unwrap();
(config_content, config_path)
```
and
```
let (config, _) = match parse_config_file_as_checkerconfig() {
    Some(config) => config,
    None => (CheckerConfig::default(), None),
};

(config, None)
```

The errors are now controlled inside the functions because it was the only way to keep our current logic intact (and it's probably cleaner too).